server/service/service_osquery.go

```diff
@@ -700,3 +702,33 @@ func (svc *Service) SubmitDistributedQueryResults(
 
 	return nil
 }
+
+func logJSON(logger log.Logger, v interface{}, key string) {
+	jsonV, err := json.Marshal(v)
+	if err != nil {
+		level.Debug(logger).Log("err", errors.Wrapf(err, "marshaling %s for debug", key))
+		return
+	}
+	level.Debug(logger).Log(key, string(jsonV))
+}
+
+func (svc *Service) maybeDebugHost(
+	host fleet.Host,
+	results fleet.OsqueryDistributedQueryResults,
+	statuses map[string]fleet.OsqueryStatus,
+	messages map[string]string,
+) {
+	hlogger := log.With(svc.logger, "host-id", host.ID)
+	ac, err := svc.ds.AppConfig()
+	if err != nil {
+		level.Debug(hlogger).Log("err", errors.Wrap(err, "getting app config for host debug"))
+		return
+	}
+
+	if strings.Contains(ac.ServerSettings.DebugHostIDs, fmt.Sprint(host.ID)) {
```

Comment:
The debug host IDs config being a simple string, that would enable debugging in cases like we have `123, 456` for hosts with ID `1`, `2`, etc. A safer way would be to split the DebugHostIDs on the comma, trim the spaces and check if the host id matches exacty.

Reply:
That's true, didn't think it through too much. Will improve.